style: apply /style-guide pass to models/sweeps (#2722)
wandb/docs: models/sweeps/initialize-sweeps.mdx

---
description: "Initialize a sweep using the Python SDK or CLI to start hyperparameter searches with your sweep configuration."
title: Initialize a sweep
---

> Initialize a sweep to register your sweep configuration with W&B and return a sweep ID that agents use to fetch run instructions. Initialize a sweep after you define a sweep configuration and before you launch agents, so that you have a sweep ready to coordinate hyperparameter searches across one or more machines.

Reviewer: ```suggestion
After you define a sweep configuration and before you launch agents, initialize a sweep to register its configuration with W&B and receive a sweep ID. Agents use the sweep ID to fetch run instructions. After a sweep is initialized, it's ready to coordinate hyperparameter searches across one or more machines.
```

Author: Applied your suggestion.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

> W&B uses a _sweep controller_ to manage sweeps on the cloud (standard) or locally across one or more machines. After a run completes, the sweep controller issues a new set of instructions describing a new run to execute. _Agents_ pick up these instructions and perform the runs. In a typical sweep, the controller lives on the W&B server. Agents live on your machines.

Reviewer: Not sure the italics are needed. We don't usually talk about "on the cloud" and "locally" and "Server" in the main docs. I wonder if it can be rewritten a bit.

Author: Rewrote to drop the italics and the "on the cloud (standard) or locally ... Server" phrasing: "W&B uses a sweep controller to manage sweeps across one or more machines... By default, W&B hosts the controller, and agents run on your machines."

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

sweep_id = wandb.sweep(sweep=sweep_configuration, project="project-name")
```

> The [`wandb.sweep()`](/models/ref/python/functions/sweep) function returns the sweep ID. The sweep ID includes the entity name and the project name. Note the sweep ID, since you pass it to agents when you start them. With the sweep ID, you're ready to launch one or more agents to execute runs.

Reviewer: ```suggestion
The [`wandb.sweep()`](/models/ref/python/functions/sweep) function returns the sweep ID. The sweep ID includes the entity name and the project name. Note the sweep ID, which you pass to agents when you start them. With the sweep ID, you're ready to launch one or more agents to execute runs.
```

Author: Applied your suggestion ("since" → "which"). Also applied the same fix to the identical sentence in the CLI tab for consistency.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

</Tab>
<Tab title="CLI">

> Use the W&B CLI to initialize a sweep. Provide the name of your configuration file. Optionally provide the name of the project for the `project` flag. If you don't specify the project, W&B puts the run in an "Uncategorized" project.

Reviewer: ```suggestion
Use the W&B CLI to initialize a sweep. Provide the name of your configuration file. Optionally set the `--project` flag to the project name. If you don't specify the project, W&B puts the run in an "Uncategorized" project.
```

Author: Applied your suggestion (set the `--project` flag to the project name). Also made the Python tab's project sentence parallel ("set the `project` parameter to the project name").

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
